ci/linux-packages/apt/debian-bullseye/Dockerfile

```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-ARG FROM=debian:bullseye
+ARG FROM=golang:1.21-bullseye
```

Comment:
Isn't the point of this pipeline to test a (reasonably) normal Debian install? This image looks to just download Go from the website. (What do we do for upstream? I don't see Go in the [upstream Dockerfile](https://github.com/apache/arrow/blob/main/dev/tasks/linux-packages/apache-arrow/apt/debian-bullseye/Dockerfile))

CC @kou

Reply:
I'm having a helluva time trying to get the golang-1.21 backports installed in this dockerfile and used properly for packaging. 

I second the request for anyone (@kou) to help with this little packaging dilemma I'm running into.... :frowning:

Reply:
Sorry. I missed this.

We can use `golang:1.21-bullseye` here or we can drop support for Debian Bullseye. Because Debian Bullseye will reach EOL on 2024-07: https://wiki.debian.org/DebianReleases

Reply:
Which is preferred? Can we drop support for Debian Bullseye?

Reply:
I'm OK if we want to drop Bullseye.

Reply:
OK. I'll drop support for Bullseye.